### PR TITLE
add utility functions test

### PR DESCRIPTION
**Pull request by kotarella1110**

### CODE REVIEW

File: src/utils/set.test.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import set from './set';
+
+describe('set', () => {
+  it('should set the correct values', () => {
+    const test1 = { a: [{ b: { c: 3 } }] };
+    expect(set(test1, 'a[0].b.c', 4)).toEqual(4);
+    expect(test1.a[0].b.c).toEqual(4);
+
+    const test2 = { foo: { bar: 'baz' } };
+    expect(set(test2, 'foo.arr[0]', 3)).toEqual(3);
```
Comment (kotarella1110) [inline src/utils/set.test.ts:10]: @bluebill1049 Is this the expected return value?
I was expecting the following return value.

```ts
expect(set(test2, 'foo.arr[0]', 3)).toEqual({
  foo: {
    bar: 'baz',
    arr: [3],
  },
});
```

Comment (bluebill1049) [reply src/utils/set.test.ts:10]:
```
set(test2, 'foo.arr[0]', 3)
expect(test2).toEqual(3);
```
This works?

Comment (kotarella1110) [reply src/utils/set.test.ts:10]: this works

```ts
const test2 = { foo: { bar: 'baz' } };
const returnValue = set(test2, 'foo.arr[0]', 3);
expect(test2).toEqual(3); // <= expected?
expect(test2).toEqual({
  foo: {
    bar: 'baz',
    arr: [3],
  },
}); // correct!
```

Comment (bluebill1049) [reply src/utils/set.test.ts:10]: `expect(returnValue).toEqual(3);` expected because this method mutates the original object.

**Review by bluebill1049 — APPROVED** [review]
awesome! thanks @kotarella1110  for refining this lib with more tests. 🙏

**Pull request merged**